ibtk/src/utilities/CellNoCornersFillPattern.cpp

```diff
@@ -132,68 +69,34 @@ CellNoCornersFillPattern::calculateOverlapOnLevel(const BoxGeometry<NDIM>& dst_g
                                                   const int dst_level_num,
                                                   const int /*src_level_num*/) const
 {
-    Pointer<CellOverlap<NDIM> > box_geom_overlap =
-        dst_geometry.calculateOverlap(src_geometry, src_mask, overwrite_interior, src_offset);
-#if !defined(NDEBUG)
-    TBOX_ASSERT(box_geom_overlap);
-#endif
-    auto const t_dst_geometry = dynamic_cast<const CellGeometry<NDIM>*>(&dst_geometry);
-#if !defined(NDEBUG)
-    TBOX_ASSERT(t_dst_geometry);
-#endif
-    BoxList<NDIM> dst_boxes;
-    if (!box_geom_overlap->isOverlapEmpty())
+    if (d_target_level_num == dst_level_num)
     {
-        const Box<NDIM>& dst_box = t_dst_geometry->getBox();
-        const BoxList<NDIM>& box_geom_overlap_boxes = box_geom_overlap->getDestinationBoxList();
-
-        // Determine the stencil boxes with the specified ghost cell width.
+        Pointer<CellOverlap<NDIM> > default_overlap =
+            dst_geometry.calculateOverlap(src_geometry, src_mask, overwrite_interior, src_offset);
+        auto dst_boxes = default_overlap->getDestinationBoxList();
+        auto dst_cell_box = CellGeometry<NDIM>::toCellBox(dst_patch_box);
         BoxList<NDIM> stencil_boxes;
-        if (NDIM == 2 || (!d_include_edges_on_dst_level && dst_level_num == d_target_level_num) ||
-            (!d_include_edges_on_src_level && dst_level_num != d_target_level_num))
+        if (d_overwrite_interior) stencil_boxes.addItem(dst_cell_box);
```

Comment:
What is the difference between the member variable `d_overwrite_interior` and function variable `overwrite_interior`?